model/data_retention_policy.go

```diff
@@ -8,20 +8,101 @@ import (
 	"io"
 )
 
-type DataRetentionPolicy struct {
+type GlobalRetentionPolicy struct {
 	MessageDeletionEnabled bool  `json:"message_deletion_enabled"`
 	FileDeletionEnabled    bool  `json:"file_deletion_enabled"`
 	MessageRetentionCutoff int64 `json:"message_retention_cutoff"`
 	FileRetentionCutoff    int64 `json:"file_retention_cutoff"`
 }
 
-func (drp *DataRetentionPolicy) ToJson() string {
-	b, _ := json.Marshal(drp)
+type RetentionPolicy struct {
+	Id           string `json:"id"`
+	DisplayName  string `json:"display_name"`
+	PostDuration int64  `json:"post_duration"`
+}
+
+type ChannelDisplayInfo struct {
+	Id              string `json:"id"`
+	DisplayName     string `json:"display_name"`
+	TeamDisplayName string `json:"team_display_name"`
+}
+
+type TeamDisplayInfo struct {
+	Id          string `json:"id"`
+	DisplayName string `json:"display_name"`
+}
+
+type RetentionPolicyEnriched struct {
+	Id           string               `json:"id"`
+	DisplayName  string               `json:"display_name"`
+	PostDuration int64                `json:"post_duration"`
+	Teams        []TeamDisplayInfo    `json:"teams"`
+	Channels     []ChannelDisplayInfo `json:"channels"`
+}
+
+type RetentionPolicyUpdate struct {
+	Id           string   `json:"id"`
+	DisplayName  string   `json:"display_name"`
+	PostDuration int64    `json:"post_duration"`
+	TeamIds      []string `json:"team_ids"`
+	ChannelIds   []string `json:"channel_ids"`
+}
+
+type RetentionPolicyWithCounts struct {
+	Id           string `json:"id"`
+	DisplayName  string `json:"display_name"`
+	PostDuration int64  `json:"post_duration"`
+	ChannelCount int64  `json:"channel_count"`
+	TeamCount    int64  `json:"team_count"`
+}
+
+type RetentionPolicyChannel struct {
+	PolicyId  string
+	ChannelId string
+}
+
+type RetentionPolicyTeam struct {
+	PolicyId string
+	TeamId   string
+}
+
+func (grp *GlobalRetentionPolicy) ToJson() string {
+	b, _ := json.Marshal(grp)
 	return string(b)
 }
 
-func DataRetentionPolicyFromJson(data io.Reader) *DataRetentionPolicy {
-	var drp *DataRetentionPolicy
-	json.NewDecoder(data).Decode(&drp)
-	return drp
+func GlobalRetentionPolicyFromJson(data io.Reader) *GlobalRetentionPolicy {
+	var grp *GlobalRetentionPolicy
+	json.NewDecoder(data).Decode(&grp)
+	return grp
+}
+
+func (rp *RetentionPolicy) ToJsonBytes() []byte {
+	b, _ := json.Marshal(rp)
+	return b
+}
+
+func (rp *RetentionPolicy) ToJson() string {
+	return string(rp.ToJsonBytes())
+}
+
+func (rp *RetentionPolicyEnriched) ToJsonBytes() []byte {
+	b, _ := json.Marshal(rp)
+	return b
+}
+
+func (rp *RetentionPolicyEnriched) ToJson() string {
+	return string(rp.ToJsonBytes())
+}
```

Comment:
It doesn't appear you ever use a `ToJson()` as a string. When it is used, it gets cast back to a byte[]. I suggest creating just a `ToJson()` and eliminating `ToJsonBytes()`.